bin/codebasin

```diff
@@ -189,9 +202,14 @@ def main():
             )
         filtered_platforms.append(p)
 
-    # If no additional platforms are specified, a config file is required.
+    # A legacy config file is required if:
+    # - No additional platforms are specified; or
+    # - No TOML analysis file is specified
     config_file = args.config_file
-    if len(additional_platforms) == 0 and config_file is None:
+    config_required = (
+        len(additional_platforms) == 0 and args.analysis_file is None
+    )
+    if config_file is None and config_required:
```

Comment:
Do we want to have a warning message here too, just to let the user know we are defaulting to YAML instead of silently?

Reply:
This branch assigns to `config_file`, so the warning on Line 229 already triggers if you implicitly use a YAML file:
```
$ codebasin

codebasin:229: UserWarning: YAML configuration files are deprecated. Use TOML analysis files instead.
```

Reply:
Yeah but it doesn't tell you why that message is being thrown: if you didn't know you'd ask why is it telling me this right?

Reply:
True.  You've convinced me; I've added a separate warning in 0fdaab616e5bc63b0f5b536bb95596c5abf5e738:

```sh
$ codebasin

codebasin:213: UserWarning: Implicitly defined configuration files are deprecated.
codebasin:232: UserWarning: YAML configuration files are deprecated. Use TOML analysis files instead.
```